hs_core/models.py

```diff
@@ -961,6 +961,36 @@ def __unicode__(self):
         return self.value
 
 
+class FundingAgency(AbstractMetaDataElement):
+    term = 'FundingAgency'
+    agency_name = models.TextField(null=False)
+    award_title = models.TextField(null=True, blank=True)
+    award_number = models.TextField(null=True, blank=True)
+    agency_url = models.URLField(null=True, blank=True)
+
+    class Meta:
+        unique_together = ("content_type", "object_id", "award_number", "agency_name")
```

Comment:
@hyi By having 'award_number' it gives me the desired uniqueness, We can have 2 or more funding agencies with same name as long as the award numbers are different. In case of award_number being null, we can have multiple finding agencies with the same name. If we need to prevent multiple funding agencies with same name but no award_number (null) then I have to add some validation code for that.